### PR TITLE
[ie/Dropbox] Fix extractor

### LINKED ISSUE #7005: [Dropbox] Unable to extract mountComponent
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting that a **supported** site is broken
- [X] I've verified that I'm running yt-dlp version **2023.03.04** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/yt-dlp/yt-dlp/wiki/FAQ#video-url-contains-an-ampersand--and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [ ] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

_No response_

### Provide a description that is worded well enough to be understood

hello,
cannot extract from dropbox.com. demo link [NSFW]: REDACTED

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [ ] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
C:\Users\MASTERX>youtube-dl "REDACTED" -FUv
[debug] Command-line config: ['REDACTED', '-FUv']
[debug] Encodings: locale cp1252, fs utf-8, pref cp1252, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2023.03.04 [392389b7d] (win_exe)
[debug] Python 3.8.10 (CPython AMD64 64bit) - Windows-10-10.0.19045-SP0 (OpenSSL 1.1.1k  25 Mar 2021)
[debug] exe versions: ffmpeg N-110072-g073ec3b9da-20230325 (setts), ffprobe N-110072-g073ec3b9da-20230325
[debug] Optional libraries: Cryptodome-3.17, brotli-1.0.9, certifi-2022.12.07, mutagen-1.46.0, sqlite3-2.6.0, websockets-10.4
[debug] Proxy map: {}
[debug] Loaded 1786 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Available version: stable@2023.03.04, Current version: stable@2023.03.04
Current Build Hash: 5590c57bd0433ed239a2deaaf92e2ad6f37fe50f53664c821575cafe106a9421
yt-dlp is up to date (stable@2023.03.04)
[Dropbox] Extracting URL: REDACTED
[Dropbox] q9n7yl55jdub33o: Downloading webpage
ERROR: [Dropbox] q9n7yl55jdub33o: Unable to extract mountComponent; please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
  File "yt_dlp\extractor\common.py", line 694, in extract
  File "yt_dlp\extractor\dropbox.py", line 56, in _real_extract
  File "yt_dlp\extractor\common.py", line 1256, in _search_json
  File "yt_dlp\extractor\common.py", line 1242, in _search_regex
```

### LINKED ISSUE #7696: [dropbox] download unsupported url
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting that yt-dlp is broken on a **supported** site
- [X] I've verified that I'm running yt-dlp version **2023.07.06** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/yt-dlp/yt-dlp/wiki/FAQ#video-url-contains-an-ampersand--and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

America

### Provide a description that is worded well enough to be understood

i share a dropbox file with link https://www.dropbox.com/scl/fi/ma1u2bi1v192j5nlj1mff/7-30min-1080p-2.15G.mov?rlkey=4zoz42qxs9knp9hvzazig1a8a&dl=0, but when i try to download it, yt-dlp display unsupported link error.
i create this share link by 
image
is there other way to create dropbox's support link?

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [ ] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2023.07.06 [b532a3481] (pip)
[debug] Python 3.10.10 (CPython x86_64 64bit) - Linux-5.19.0-1022-gcp-x86_64-with-glibc2.35 (OpenSSL 3.0.2 15 Mar 2022, glibc 2.35)
[debug] exe versions: ffmpeg 4.4.2 (setts), ffprobe 4.4.2, phantomjs broken
[debug] Optional libraries: Cryptodome-3.17, brotli-1.0.9, certifi-2022.12.07, mutagen-1.46.0, sqlite3-2.6.0, websockets-11.0.2
[debug] Proxy map: {}
[debug] Loaded 1855 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Available version: stable@2023.07.06, Current version: stable@2023.07.06
yt-dlp is up to date (stable@2023.07.06)
[generic] Extracting URL: https://www.dropbox.com/scl/fi/ma1u2bi1v192j5nlj1mff/7-30min-1080p-2.15G.mov?rlkey=4zoz42qxs9knp9hvzazig1a8a
[generic] 7-30min-1080p-2.15G: Downloading webpage
[redirect] Following redirect to https://www.dropbox.com/e/scl/fi/ma1u2bi1v192j5nlj1mff/7-30min-1080p-2.15G.mov?rlkey=4zoz42qxs9knp9hvzazig1a8a
[generic] Extracting URL: https://www.dropbox.com/e/scl/fi/ma1u2bi1v192j5nlj1mff/7-30min-1080p-2.15G.mov?rlkey=4zoz42qxs9knp9hvzazig1a8a
[generic] 7-30min-1080p-2.15G: Downloading webpage
WARNING: [generic] Falling back on generic information extractor
[generic] 7-30min-1080p-2.15G: Extracting information
[debug] Looking for embeds
ERROR: Unsupported URL: https://www.dropbox.com/e/scl/fi/ma1u2bi1v192j5nlj1mff/7-30min-1080p-2.15G.mov?rlkey=4zoz42qxs9knp9hvzazig1a8a
Traceback (most recent call last):
  File "/root/task-queue/venv/lib/python3.10/site-packages/yt_dlp/YoutubeDL.py", line 1560, in wrapper
    return func(self, *args, **kwargs)
  File "/root/task-queue/venv/lib/python3.10/site-packages/yt_dlp/YoutubeDL.py", line 1688, in __extract_info
    ie_result = ie.extract(url)
  File "/root/task-queue/venv/lib/python3.10/site-packages/yt_dlp/extractor/common.py", line 710, in extract
    ie_result = self._real_extract(url)
  File "/root/task-queue/venv/lib/python3.10/site-packages/yt_dlp/extractor/generic.py", line 2568, in _real_extract
    raise UnsupportedError(url)
yt_dlp.utils.UnsupportedError: Unsupported URL: https://www.dropbox.com/e/scl/fi/ma1u2bi1v192j5nlj1mff/7-30min-1080p-2.15G.mov?rlkey=4zoz42qxs9knp9hvzazig1a8a
```

### PR DESCRIPTION
**Pull request by nathantouze**
Since dropbox updated their share links around July this year, I had to make an update it to keep using this extractor. 

This new regex handles old and new link formats.

Dropbox changed their link format earlier this year. The exctractor weren't working anymore with new links. A change was mandatory to keep using the software. 
Old formats: 
- `https://www.dropbox.com/s/<id>/* `
- `https://www.dropbox.com/sh/<id>/*`

Added formats:
-  `https://www.dropbox.com/e/scl/fi/<id>/*`
-  `https://www.dropbox.com/scl/fi/<id>/*`

(EDIT: Also fixed extraction --@bashonly)

Closes #7005, Closes #7696

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check all of the following options that apply:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [x] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence) @denhotte and @bashonly are responsible for changes to format extraction code (https://github.com/yt-dlp/yt-dlp/issues/7005#issuecomment-1613531558)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

</details><details><summary>Copilot Summary</summary>  

### Summary
📁🔗🎥

Improve Dropbox extractor to support more link formats. Update `_VALID_URL` in `yt_dlp/extractor/dropbox.py` to match them.

> _Dropbox links vary_
> _`_VALID_URL` adapts_
> _Winter of patterns_

### Walkthrough
* Expand the `_VALID_URL` regex pattern to support more Dropbox link formats ([link](https://github.com/yt-dlp/yt-dlp/pull/7926/files?diff=unified&w=0#diff-ee2d05a5425abc12f101f94eb4915479ed5315ae7aebc36e0f4b8097aebb0f6eL15-R15))

</details>

### CODE REVIEW

**Comment by bashonly** [comment]
Add a test for each of the new URL formats

File: yt_dlp/extractor/dropbox.py [hunk 1]
```diff
@@ -12,7 +12,7 @@
 
 
 class DropboxIE(InfoExtractor):
-    _VALID_URL = r'https?://(?:www\.)?dropbox[.]com/sh?/(?P<id>[a-zA-Z0-9]{15})/.*'
+    _VALID_URL = r'https?://(?:www\.)?dropbox\.com/(?:scl/fi/|e/scl/fi/|s/|sh/)(?P<id>[^/]+)/.*'
```
Comment (bashonly) [inline yt_dlp/extractor/dropbox.py:15]: isn't `\w+` enough for the `id`?
```suggestion
    _VALID_URL = r'https?://(?:www\.)?dropbox\.com/(?:(?:e/)?scl/fi|sh?)/(?P<id>\w+)/.*'
```
or else this:
```suggestion
    _VALID_URL = r'https?://(?:www\.)?dropbox\.com/(?:(?:e/)?scl/fi|sh?)/(?P<id>[^/#?]+)/.*'
```

**Review by bashonly — CHANGES_REQUESTED** [review]

**Comment by DoubiTe** [comment]
@bashonly & @nathantouze this PR is partially fix the extractor because it still generate an error "mountComponent" #7005
Is it possible to fix-it @pukkandan ? Thanks a lot

**Review by DoubiTe — APPROVED** [review]

**Review by DoubiTe — APPROVED** [review]

File: yt_dlp/extractor/dropbox.py [hunk 2]
```diff
@@ -22,9 +22,17 @@ class DropboxIE(InfoExtractor):
...
         },
+        {
+            "url": "https://www.dropbox.com/scl/fi/r2kd2skcy5ylbbta5y1pz/DJI_0003.MP4?dl=0&rlkey=wcdgqangn7t3lnmmv6li9mu9h",
+            "only_matching": True,
+        },
+        {
+            "url": "https://www.dropbox.com/e/scl/fi/r2kd2skcy5ylbbta5y1pz/DJI_0003.MP4?dl=0&rlkey=wcdgqangn7t3lnmmv6li9mu9h",
+            "only_matching": True,
+        }
     ]
```
Comment (bashonly) [inline yt_dlp/extractor/dropbox.py:36]:
```suggestion
        }, {
            'url': 'https://www.dropbox.com/scl/fi/r2kd2skcy5ylbbta5y1pz/DJI_0003.MP4?dl=0&rlkey=wcdgqangn7t3lnmmv6li9mu9h',
            'only_matching': True,
        }, {
            'url': 'https://www.dropbox.com/e/scl/fi/r2kd2skcy5ylbbta5y1pz/DJI_0003.MP4?dl=0&rlkey=wcdgqangn7t3lnmmv6li9mu9h',
            'only_matching': True,
        },
    ]
```

**Review by bashonly — APPROVED** [review]

File: yt_dlp/extractor/dropbox.py [hunk 3]
```diff
@@ -1,18 +1,18 @@
...
-    traverse_obj,
-    try_get,
+    update_url_query,
     url_basename,
 )
 
 
 class DropboxIE(InfoExtractor):
-    _VALID_URL = r'https?://(?:www\.)?dropbox[.]com/sh?/(?P<id>[a-zA-Z0-9]{15})/.*'
+    _VALID_URL = r'https?://(?:www\.)?dropbox\.com/(?:(?:e/)?scl/fi|sh?)/(?P<id>\w+)/.*'
```
Comment (pukkandan) [inline yt_dlp/extractor/dropbox.py:15]:
```suggestion
    _VALID_URL = r'https?://(?:www\.)?dropbox\.com/(?:(?:e/)?scl/fi|sh?)/(?P<id>\w+)'
```

Example: `https://www.dropbox.com/s/nelirfsxnmcfbfh`

**Review by pukkandan — CHANGES_REQUESTED** [review]

**Pull request merged**